src/sdk/wallet/providers/wallet-connect-2.wallet-provider.ts

```diff
@@ -0,0 +1,60 @@
+import { BytesLike } from 'ethers';
+import { toHex } from '../../common';
+import { DynamicWalletProvider } from './dynamic.wallet-provider';
+import { EthereumProvider } from './interfaces';
+
+export class WalletConnect2WalletProvider extends DynamicWalletProvider {
+  constructor(readonly provider: EthereumProvider) {
+    super('WalletConnect2');
+
+    try {
+      const {
+        accounts: [address],
+        chainId,
+      } = provider;
+
+      this.setAddress(address);
+      this.setNetworkName(chainId);
+    } catch (err) {
+      //
+    }
+
+    this.updateSessionHandler = this.updateSessionHandler.bind(this);
+
+    provider.on('connect', this.updateSessionHandler);
+    provider.on('session_event', this.updateSessionHandler);
+    provider.on('disconnect', () => {
+      this.setAddress(null);
+      this.setNetworkName(null);
+    });
+  }
+
+  async signMessage(message: BytesLike): Promise<string> {
+    const response = await this.provider.signer.request({
+      method: 'personal_sign',
+      params: [toHex(message), this.address],
+    });
```

Comment:
Just wondering if htis could return an error from the `await`?

Reply:
this is copy paste implementation of previous Wallet Connect provider, it can return error, but it wasn't handled in previous Wallet Connect provider so didn't feel to add anything extra that is beyond previous structure